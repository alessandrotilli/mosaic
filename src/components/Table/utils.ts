export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
