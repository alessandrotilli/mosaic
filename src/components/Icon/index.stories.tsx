import React from "react";
import MUIStyleIcon from "@mui/icons-material/Style";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Icons } from "../../types/Icon";
import getDocsPage from "../../utils/stories";

import Icon, { DATA_CY_DEFAULT } from ".";

export default {
  title: "Display/Icon",
  component: Icon,
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Icon Component",
          url: "https://v4.mui.com/components/icons/",
        },
        component: "Icon",
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
        },
      }),
    },
  },
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} dataCy={DATA_CY_DEFAULT} />;

export const Primary = Template.bind({});
Primary.args = {
  name: Icons.account,
};

export const IconCustom = Template.bind({});
IconCustom.args = {
  children: <MUIStyleIcon />,
};

export const Loading = Template.bind({});
Loading.args = {
  ...Primary.args,
  loading: true,
};

export const Rotate = Template.bind({});
Rotate.args = {
  ...Primary.args,
  rotate: true,
};

export const SizeSmall = Template.bind({});
SizeSmall.args = {
  ...Primary.args,
  size: "small",
};

export const SizeLarge = Template.bind({});
SizeLarge.args = {
  ...Primary.args,
  size: "large",
};

export const Styled = Template.bind({});
Styled.args = {
  ...Primary.args,
  style: {
    backgroundColor: "red",
    borderRadius: "4px",
    color: "white",
    padding: "4px",
  },
};

// TODO: add this story using union type
// const allIcons = Object.values(Icons);
// const getIcon = (icon: string) => allIcons.find((i) => i === icon);
// const allIconsJsx = allIcons.map((icon, index) => (
//   <div key={index} className="icon-wrapper">
//     <Icon dataCy={`icon-${icon}`} name={getIcon(icon) || Icons.add} />
//     <span>{icon}</span>
//   </div>
// ));

// export const AllIcons = () => <StoriesWrapper>{allIconsJsx}</StoriesWrapper>;
