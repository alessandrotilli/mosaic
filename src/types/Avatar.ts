import { ILoadable } from "./Base";
import { Icons } from "./Icon";

export enum AvatarVariant {
  default = "circular",
  rounded = "rounded",
  squared = "square",
}

export interface IAvatar extends ILoadable {
  alt?: string;
  icon?: Icons;
  src?: string;
  text?: string;
  variant?: AvatarVariant;
}
