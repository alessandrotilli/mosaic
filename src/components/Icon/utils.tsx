import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
