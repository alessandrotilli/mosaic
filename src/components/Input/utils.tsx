import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
