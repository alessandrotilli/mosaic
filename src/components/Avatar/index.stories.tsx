import React from "react";
import MUIStyleIcon from "@mui/icons-material/Style";
import { Meta, StoryObj } from "@storybook/react";

import { Icons } from "../../types/Icon";
import { getAllComposedDataCy } from "../../utils";
import getDocsPage from "../../utils/stories";

import Avatar, { DATA_CY_DEFAULT, SUBPARTS_MAP } from ".";

const meta = {
  title: "Display/Avatar",
  component: Avatar,
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Avatar Component",
          url: "https://mui.com/components/avatars/",
        },
        component: "Avatar",
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
          subpartsSuffixes: getAllComposedDataCy(SUBPARTS_MAP),
        },
      }),
    },
  },
} satisfies Meta<typeof Avatar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    icon: Icons.add,
  },
};

export const CustomIcon: Story = {
  args: {
    ...Primary.args,
    icon: <MUIStyleIcon />,
  },
};

export const Image: Story = {
  args: {
    src: "//upload.wikimedia.org/wikipedia/commons/thumb/6/60/Roof_hafez_tomb.jpg/440px-Roof_hafez_tomb.jpg",
  },
};

export const Loading: Story = {
  args: {
    loading: true,
  },
};

export const LoadingVariant: Story = {
  args: {
    loading: true,
    variant: "square",
  },
};

export const Styled: Story = {
  args: {
    ...Primary.args,
    style: {
      backgroundColor: "red",
      color: "white",
    },
  },
};

export const Text: Story = {
  args: {
    text: "MO",
  },
};

export const VariantRounded: Story = {
  args: {
    ...Primary.args,
    variant: "rounded",
  },
};

export const VariantSquare: Story = {
  args: {
    ...Primary.args,
    variant: "square",
  },
};
