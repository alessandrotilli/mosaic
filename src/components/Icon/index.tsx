import React, { cloneElement, FC, PropsWithChildren, ReactElement, useCallback, useMemo } from "react";
import { ClassNames } from "@emotion/react";
import { keyframes, Skeleton as MUISkeleton } from "@mui/material";

import { IIcon, IIconDimensions, IRenderedIcon } from "../../types/Icon";
import { logWarn } from "../../utils/logger";
import Adornment from "../Adornment";

import { iconsCatalog } from "./utils";

export const DATA_CY_DEFAULT = "icon";

const ICON_DIMENSIONS: IIconDimensions = {
  small: 20,
  medium: 24,
  large: 35,
};

const Icon: FC<PropsWithChildren<IIcon>> = ({
  badge,
  children,
  dataCy = DATA_CY_DEFAULT,
  forwarded = {},
  loading = false,
  name,
  rotate = false,
  size = "medium",
  style: externalStyle,
  tooltip,
}) => {
  const dimensions = useMemo(() => {
    const dimension = ICON_DIMENSIONS[size];
    return { height: dimension, width: dimension };
  }, [size]);

  const props = useMemo(() => {
    let props: IRenderedIcon = {
      ...forwarded,
      "data-cy": dataCy,
      fontSize: size,
    };

    if (externalStyle) {
      props = { ...props, style: { ...externalStyle } };
    }

    if (children) {
      props = { ...props, style: { ...props.style, ...dimensions } };
    }

    return props;
  }, [children, dataCy, dimensions, externalStyle, forwarded, size]);

  const rotateKeyframes = useMemo(
    () => keyframes`
      from: {
        transform: "rotate(0deg)";
      }
      to: {
        transform: "rotate(360deg)";
      }
  `,
    []
  );

  const renderWithAnimation = useCallback(
    (element: any) => {
      return (
        <ClassNames>
          {({ css }) => {
            let animation = "none";
            if (rotate) {
              animation = css`
                ${rotateKeyframes} 2s linear infinite
              `;
            }

            let className = css`
              animation: ${animation};
            `;

            if (forwarded.className) {
              className = forwarded.className;
            }

            return (
              <Adornment badge={badge} tooltip={tooltip}>
                {cloneElement(element as ReactElement<any>, { ...props, className })}
              </Adornment>
            );
          }}
        </ClassNames>
      );
    },
    [badge, forwarded, props, rotate, rotateKeyframes, tooltip]
  );

  if (loading) {
    return <MUISkeleton variant="rectangular" style={{ ...dimensions }} />;
  }

  if (children) {
    return renderWithAnimation(children);
  }

  if (!name) {
    logWarn("Icon", "Skip rendering, both children and name are not set");
    return null;
  }

  const icon = iconsCatalog[name];
  return renderWithAnimation(icon);
};

export default Icon;
