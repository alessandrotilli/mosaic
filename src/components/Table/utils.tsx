import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionPosition, ITableAction, ITableColumn } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";
import { getDataCyForSubComponent, suppressEvent } from "../../utils";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: ITableAction): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    position = TableActionPosition.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    onClick: (event, data) => {
      if (event) {
        suppressEvent(event);
      }
      callback(data);
    },
    position,
  };
};

export const actionComponentAdapter = (props: any, dataCy: string) => {
  const { data, action } = props;
  const { disabled, hidden, icon, position, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const displaysButton = position === TableActionPosition.default;
  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {displaysButton && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={() => onClick(undefined, data)}
          />
        </Fragment>
      )}
      {!displaysButton && (
        <Fragment>
          <IconButton
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={iconName}
            onClick={() => onClick(undefined, data)}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: ITableColumn): MTColumnType<object> => {
  const { label, path, render, width } = column;
  return {
    ...(!!width && ({ width } as object)),
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
