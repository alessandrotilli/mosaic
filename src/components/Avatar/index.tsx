import React, { FC, memo } from "react";
import { Skeleton as MUISkeleton } from "@material-ui/lab";

import { AvatarVariant, IAvatar } from "../../types/Avatar";
import { getComposedDataCy } from "../../utils";
import Icon from "../Icon";
import Typography from "../Typography";

import { StyledMUIAvatar } from "./styled";

export const DATA_CY_DEFAULT = "avatar";

export const SUBPARTS_MAP = {
  loading: {
    label: "Loading",
  },
  icon: {
    label: "Icon",
  },
  text: {
    label: "Text",
  },
};

const Avatar: FC<IAvatar> = ({
  alt = "avatar",
  dataCy = DATA_CY_DEFAULT,
  icon,
  loading = false,
  src,
  text,
  variant = AvatarVariant.default,
}) => {
  if (loading) {
    return (
      <MUISkeleton variant="circle">
        <StyledMUIAvatar data-cy={getComposedDataCy(dataCy, SUBPARTS_MAP.loading)} />
      </MUISkeleton>
    );
  }

  return (
    <StyledMUIAvatar alt={text || alt} data-cy={dataCy} src={src} variant={variant}>
      {icon && <Icon dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.icon)} name={icon} />}
      {!icon && text && <Typography dataCy={getComposedDataCy(dataCy, SUBPARTS_MAP.text)}>{text}</Typography>}
    </StyledMUIAvatar>
  );
};

export const AvatarWithProps = Avatar;

export default memo(Avatar);
