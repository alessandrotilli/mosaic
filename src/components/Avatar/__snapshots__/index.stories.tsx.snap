// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display/Avatar CustomIcon smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
     data-cy="avatar"
     style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="StyleIcon"
       data-cy="avatar-icon"
       style="height: 24px; width: 24px;"
  >
    <path d="m2.53 19.65 1.34.56v-9.03l-2.43 5.86c-.41 1.02.08 2.19 1.09 2.61zm19.5-3.7L17.07 3.98c-.31-.75-1.04-1.21-1.81-1.23-.26 0-.53.04-.79.15L7.1 5.95c-.75.31-1.21 1.03-1.23 1.8-.01.27.04.54.15.8l4.96 11.97c.31.76 1.05 1.22 1.83 1.23.26 0 .52-.05.77-.15l7.36-3.05c1.02-.42 1.51-1.59 1.09-2.6zM7.88 8.75c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-2 11c0 1.1.9 2 2 2h1.45l-3.45-8.34v6.34z">
    </path>
  </svg>
</div>
`;

exports[`Display/Avatar Image smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
     data-cy="avatar"
     style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
>
  <img alt="avatar"
       src="//upload.wikimedia.org/wikipedia/commons/thumb/6/60/Roof_hafez_tomb.jpg/440px-Roof_hafez_tomb.jpg"
       class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
  >
</div>
`;

exports[`Display/Avatar Loading smoke-test 1`] = `
<span class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto css-1gtw0ls-MuiSkeleton-root">
  <div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
       data-cy="avatar-loading"
  >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="PersonIcon"
    >
      <path d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z">
      </path>
    </svg>
  </div>
</span>
`;

exports[`Display/Avatar LoadingVariant smoke-test 1`] = `
<span class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-fitContent MuiSkeleton-heightAuto css-zpjta8-MuiSkeleton-root">
  <div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
       data-cy="avatar-loading"
  >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="PersonIcon"
    >
      <path d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z">
      </path>
    </svg>
  </div>
</span>
`;

exports[`Display/Avatar Primary smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
     data-cy="avatar"
     style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="avatar-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z">
    </path>
  </svg>
</div>
`;

exports[`Display/Avatar Styled smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
     data-cy="avatar"
     style="background-color: red; color: white;"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="avatar-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z">
    </path>
  </svg>
</div>
`;

exports[`Display/Avatar Text smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
     data-cy="avatar"
     style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
>
  <p class="MuiTypography-root MuiTypography-body1 css-5wwosz-MuiTypography-root"
     data-cy="avatar-text"
  >
    MO
  </p>
</div>
`;

exports[`Display/Avatar VariantRounded smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-54qi11-MuiAvatar-root"
     data-cy="avatar"
     style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="avatar-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z">
    </path>
  </svg>
</div>
`;

exports[`Display/Avatar VariantSquare smoke-test 1`] = `
<div class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-18gxrea-MuiAvatar-root"
     data-cy="avatar"
     style="background-color: rgb(25, 118, 210); color: rgb(255, 255, 255);"
>
  <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wnp694-MuiSvgIcon-root"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="AddIcon"
       data-cy="avatar-icon"
  >
    <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z">
    </path>
  </svg>
</div>
`;
